### ISSUE TITLE
[FEATURE] Add a Save Game Progress Button to Prevent Data Loss

### ISSUE THREAD

**Issue by CLee788** [COMP3122-2324-Project/Simpsyber#44]
[image_0]: https://pfst.cf2.poecdn.net/base/image/0f93ec5be3e87625416161c70373fa1988dd85249e721017cf0995b801ee51ae?w=1862&h=1158&pmaid=49847726
The current system lacks a vital functionality that allows users to preserve their game advancement. If the browser tab happens to be closed accidentally, it results in a loss of data. To tackle this issue and enhance user satisfaction, I propose the inclusion of a "Save Game Progress" button.

Issue:
Upon starting the game, users only have access to a "Back to Home" button. If they inadvertently close the browser tab or navigate away from the page, all their progress is lost, compelling them to restart the game.

Proposed Solution:
To prevent data loss and provide a seamless gaming experience, it is advised to implement a "Save Game Progress" button. This feature empowers users to save their progress at any point and resume the game from where they left off.

Integration and Design:
The "Save Game Progress" button should be positioned within the game interface, preferably near the existing navigation bar's "Back to Home" button. It should possess a visually distinctive design, ensuring users can easily locate and utilize it without confusion. 

Functionality:
When users click the "Save Game Progress" button, the system should promptly save the current progress and store it either in the browser's local storage or on the server side.

Example:
(_Before_)
<img width="956" alt="image" src="https://github.com/COMP3122-2324-Project/Simpsyber/assets/157445841/ade9f000-c9bc-4afe-a5a1-bf3c7b25f494">

(_After the implementation_)
![Xnip2024-03-04_21-21-05][image_0]By incorporating this feature, users can enjoy the convenience and assurance that their game progress is saved, even in the event of accidental browser tab closure or page departure. This enhancement significantly improves the user experience and minimizes frustration stemming from data loss.

**Comment by lhm277** [comment]
I completely agree with your points, and I would like to add some further suggestions to enhance this feature.

1. Autosave:
I had a similar experience where I accidentally closed the game and lost all my progress. It would be helpful to have an autosave functionality in addition to the manual "Save Game Progress" button. This feature would automatically save the game progress at regular intervals, providing users with peace of mind knowing that their progress is being saved even if they forget to save manually.

2. Confirmation and Feedback:
I believe that providing users with visual feedback when they successfully save their game progress is essential. A confirmation message or a small notification that briefly appears on the screen would be ideal. Such feedback assures users that their progress has been saved, enabling them to close the tab with confidence.

3. Multiple Save Slots:
As there will be varying outcomes based on the choices made during the game, we can provide users with the flexibility to experiment or revisit different stages, like incorporating multiple save slots or profiles within the "Save Game Progress" feature. This would allow users to save their progress at different points in the game or have separate saves for different playthroughs. It enhances users' overall gaming experience and encourages them to explore different strategies and outcomes.

**Comment by CHEUNG-Choi-Chun-Ifan-23041078d** [comment]
I agree with your proposed solution to add a "Save Game Progress" button to the game. This is a common feature in many games and would be a valuable addition to this game.

1. **Save Game** 
I suggest adding this button at the corner of the game, which is very similar to the game we played online, and the user can easily find the button to save the game or export the game data.

2. I**mport Game Data**
The game may have a place to let users import the game data to continue their game. This can help users play the game on different devices but using the same profile to continue their process.

**Comment by alvinlor2001** [comment]
The concept of saving game progress is fantastic! Additionally, I propose incorporating an account linking feature in the game. This functionality would allow players to connect their accounts and seamlessly save their game data, including their gaming history and records. By implementing this simple and efficient method, the game can effectively address the challenges associated with importing and saving game progress.

Here is some implementation details idea:
### **Account Creation:**
Players would have the option to create an account within the game. They could provide their email address and create a secure password to establish their unique account.

### **Account Linking:**
Once players have created an account, they can link it to their game profile. This can be done through a dedicated account settings section within the game's menu. By entering their account credentials, the player's game progress will be associated with their account.

### **Automatic Syncing:**
With the account linking established, the game would automatically sync the player's progress, history, and records to their account in real-time. This would ensure that all the latest data is securely stored and easily accessible across multiple devices.
